experiments/yarn/src/main/scala/org/apache/gearpump/experiments/yarn/client/Client.scala

```diff
@@ -51,23 +52,26 @@ trait ClientAPI {
   def getYarnConf: YarnConfiguration
   def getAppEnv: Map[String, String]
   def getAMCapability: Resource
-  def monitorAM(appContext: ApplicationSubmissionContext): Unit
-  def uploadAMResourcesToHDFS(): Unit
+  def monitor(appId: ApplicationId): Unit
+  def upload(): Boolean
+  def start(): Boolean
+  def submit(): Try[ApplicationId]
 }
 
-class Client(configuration:AppConfig, yarnConf: YarnConfiguration, yarnClient: YarnClient) extends ClientAPI {
-  import org.apache.gearpump.experiments.yarn.client.Client._
+class Client(configuration:AppConfig, yarnConf: YarnConfiguration, yarnClient: YarnClient,
+             containerLaunchContext: (String) => ContainerLaunchContext, fileSystem: FileSystem) extends ClientAPI {
   import org.apache.gearpump.experiments.yarn.Constants._
+  import org.apache.gearpump.experiments.yarn.client.Client._
 
   val LOG: Logger = LogUtil.getLogger(getClass)
   def getConfiguration = configuration
   def getEnv = getConfiguration.getEnv _
   def getYarnConf = yarnConf
-  def getFs = FileSystem.get(getYarnConf)  
+  def getFs = fileSystem
   def jarPath = new Path(getFs.getHomeDirectory, getEnv(HDFS_ROOT) + "/jars/" )
 
   val version = configuration.getEnv("version")
```

Comment:
I think we should make "version" more explicit so that we have a 'version' field that will always match version.sbt. This should be part of the build process and should probably be handled separately.